### ISSUE TITLE
Find and Select Entities without Dimensions.

### ISSUE THREAD

**Issue by Robbo1234** [eirannejad/pyRevit#696]
Similar to Find and Select Entities without Tags, I would like to find and select the Revit entities on a Revit view that haven't been Dimensioned. (Refer to screenshot)

![image](https://user-images.githubusercontent.com/8776947/65053849-8605e100-d964-11e9-9508-221bd7575e60.png)

**Comment by htlcnn** [comment]
I'm thinking of following workflow:
- Use `FilteredElementCollector` to collect elements `OfClass(DB.Dimension)` in `doc.ActiveView`
- for each dimension: get its `References`
- for each Reference: get its `ElementId`
- use another `FilteredElementColelctor` to get all elements in `doc.ActiveView`, compare their `Id`s to `ElementId`s of `Reference`s above

Hope this helps :)

**Comment by htlcnn** [comment]
```python
"""Selects elements with no associated dimensions in current view."""
#pylint: disable=import-error,invalid-name
from System.Collections.Generic import List
from pyrevit import revit, DB
from pyrevit import forms

# make sure active view is not a sheet
curview = revit.active_view
if isinstance(curview, DB.ViewSheet):
    forms.alert("You're on a Sheet. Activate a model view please.",
                exitscript=True)


categories = {
    'Rooms': DB.BuiltInCategory.OST_Rooms,
    'Areas': DB.BuiltInCategory.OST_Areas,
    'Spaces': DB.BuiltInCategory.OST_MEPSpaces,
    'Doors': DB.BuiltInCategory.OST_Doors,
    'Windows': DB.BuiltInCategory.OST_Windows,
    'Speciality Equipment': DB.BuiltInCategory.OST_SpecialityEquipment,
    'Mechanical Equipment': DB.BuiltInCategory.OST_MechanicalEquipment,
    'Electrical Equipment': DB.BuiltInCategory.OST_ElectricalEquipment,
    'Walls': DB.BuiltInCategory.OST_Walls,
    'Curtain Walls': DB.BuiltInCategory.OST_CurtainWallPanels,
    'Ceilings': DB.BuiltInCategory.OST_Ceilings,
    'Columns': DB.BuiltInCategory.OST_StructuralColumns,
    }

selected_switch = \
    forms.CommandSwitchWindow.show(sorted(categories.keys()),
                                   message='Find undimmed elements of category:')


if selected_switch:
    target = categories[selected_switch]
    selection = revit.get_selection()
    c = DB.FilteredElementCollector(revit.doc, curview.Id)\
        .OfCategory(target)\
        .WhereElementIsNotElementType()
    all_ids = set(eid.IntegerValue for eid in c.ToElementIds())

    dim_collector = DB.FilteredElementCollector(revit.doc, curview.Id)\
                    .OfClass(DB.Dimension)

    dimmed_ids = set()
    for dim in dim_collector:
        for ref in dim.References:
            dimmed_ids.add(ref.ElementId.IntegerValue)

    not_dimmed_ids = List[DB.ElementId]()
    for eid in all_ids.difference(dimmed_ids):
        not_dimmed_ids.Add(DB.ElementId(eid))

    if not_dimmed_ids:
        revit.uidoc.Selection.SetElementIds(not_dimmed_ids)
    else:
        forms.alert('All %s have associated dimensions.' % selected_switch)
```

My implementation based on existing pyRevit button

**Comment by Robbo1234** [comment]
Thanks, @htlcnn, and @eirannejad

**Comment by eirannejad** [comment]
Added. Thanks a lot @htlcnn !! Will be published with 4.7 final

**Comment by Robbo1234** [comment]
Thanks
